init/Kconfig

```diff
@@ -2181,3 +2181,20 @@ config ARCH_HAS_SYNC_CORE_BEFORE_USERMODE
 # <asm/syscall_wrapper.h>.
 config ARCH_HAS_SYSCALL_WRAPPER
 	def_bool n
+
+config HAS_RUST
+        def_bool $(success,rustc --version)
+
+config HAS_CARGO
+        def_bool $(success,cargo --version)
+
+config HAS_CARGO_XBUILD
+        def_bool $(success,cargo xbuild --version)
+
+config RUST
+        bool
+        depends on HAS_RUST
+        depends on HAS_CARGO
+        depends on HAS_CARGO_XBUILD
+        help
+          Whether to support building modules written in Rust.
```

Comment:
Even though you autodetect the availability of Rust and Cargo, for the moment at least the request from upstream was to make sure this wasn't enabled by `make allyesconfig` or `make allmodconfig`. (For instance, just because a developer has rust and cargo installed for other reasons, doesn't necessarily mean they want their kernel tests to build this for now.)

To avoid that, I'd suggest the following pattern, inspired by a similar approach and requirement for link-time optimization (LTO). Rename `config RUST` to `config RUST_MENU`, and then add the following below it:

```
config RUST_DISABLE
	bool
	depends on RUST_MENU
	help
	  This option re-disables the support for Rust, so that make 
	  allyesconfig and make allmodconfig will not enable it. To 
	  build modules written in Rust, leave this option set to 'n'.

config RUST
	bool
	default y
	depends on RUST_MENU && !RUST_DISABLE
```

Reply:
Thanks! I wasn't sure what the right pattern was here.

Reply:
Just using "depends on !COMPILE_TEST" is sufficient.